### DISCUSSION TITLE
Completed Build

### DISCUSSION THREAD

**Discussion by mjparme** [ZeroZeroOne-dev/00Key#12] [Show and tell]
I finally finished the build and my son really likes it.  I went my own direction with the bottom "no case" and printed it with clear PETG which gives a nice RGB underglow effect similar to acrylic sandwich-style cases. It has a 7% slope. I lifted the PCB off of it with some 2.5 mm spacers which were just enough to clear the solder/components sticking out the underside of the PCB and the wires to the RGB lights. The PCB is mounted to the case with M2 screws in M2x4x3.5 heat set inserts.

I will be sharing my case design via a PR in case you want to include it in your repository.

If I could offer a couple of constructive suggestions:

1. The Teensy 4.1 is kind of pricey (\~$27) and is way overkill (600mhz and 1MB of RAM). Maybe something like an Elite-C (\~$17) or a stm32 (\~$4) would be a better choice for controller.
2. Add polarity markings for the LED legs to the PCB.

Great design!

Moneyshot:

![IMG_0881](https://user-images.githubusercontent.com/1580996/149826598-a80cf5e0-d927-4c5a-9d70-7ee13b082f6e.jpg)

Top down view with no backlight on (RGB is on which can be seen shining through the PCB a bit). If I had known my son would want green keycaps I would have had the PCB made white instead of green. However, my son said he likes the light green on the dark green so that is all that matters:

![IMG_0872](https://user-images.githubusercontent.com/1580996/149827125-36133d53-84e7-4a1b-8e1c-a92ca3cec282.jpg)

Top down view with backlight on (note my keycap set doesn't have a 1.75u shift key or 1u alt, ctrl, and cmd keys for right side so just kind of improvised). The LED doesn't work on one key, looks like I damaged the traces leading to the LED somehow, don't recall doing it but looks like I did:

![IMG_0877](https://user-images.githubusercontent.com/1580996/149829883-d4d10182-13c2-44a2-836e-5c0f72162f1d.jpg)

Under side of my case to show the RGB lights. I taped them onto the bottom of the PCB with kapton tape and then positioned them to show through the slots. This isn't the final version of the case, after mounting it noticed that middle section flexes a bit because of the slots for the RGB so now have a version with an anti-flex support running across the middle (I will share that version):

![IMG_0878](https://user-images.githubusercontent.com/1580996/149828673-ec67f00a-ca7d-4940-ab1f-7d264bc54348.jpg)

With anti-flex support:

![Screen Shot 2022-01-17 at 1 48 22 PM](https://user-images.githubusercontent.com/1580996/149829490-c32593b6-924f-492a-b70a-a44169390f79.png)

**Comment by ZeroZeroOne-dev** [comment]
Wow I really love your take on the case! Very clean build.
I actually really like how the backlit shines through the PCB, a nice green glow that goes great with the green keycaps. 

I definitely would like a PR for the case, thanks! 

As for your suggestions, I agree with all of them.

One of my current projects is trying out some dedicated micro controller designs (but the current ship shortage is making it a bit ugly). 
I chose the teensy initially because it was one of the few devboards that had enough pins without needing IO expanders / demuxers / ... Definitely not a decision I would make again now. (Although I do love the teensy lineup).

Glad you used my design, and hope your son enjoys using the board 😄

Edit: also that dual color OLED works almost perfectly with the board name, I didn't plan for that but now I might need to get one 👀

**Reply by mjparme** [reply]
> Edit: also that dual color OLED works almost perfectly with the board name, I didn't plan for that but now I might need to get one 


I have some mono ones and dual color ones. I just kind of grabbed one out of my drawer and didn't pay much attention to which one it was. Just made sure the pins were in the correct order. But now since you mention that I noticed the OLED.cpp just uses the default color for the "OOKey v1" text which must be yellow on this screen and it certainly does look intentional to have the name a different color :-)

I went on an electronics buying spree from AliExpress a couple of years back with plans on doing a whole bunch of Arduino and/or raspberry pi sensor stuff. Haven't quite gotten around to it, but that means i have tons of components and when a project like this comes up I pretty much always have the part I need and can just reach in and grab one. I probably have 7 or 8 more of these little screens. Procrastination does pay off on occasion!

**Comment by ZeroZeroOne-dev** [comment]
Oh also one last tip: You can bend the mosfet in a way that it doesnt interfere with the usb

![IMG_3550](https://user-images.githubusercontent.com/254635/149832902-e1904c73-ffd6-47bc-816a-3fada26f66ef.jpeg)

**Reply by mjparme** [reply]
Thanks for the tip! I was thinking about just standing it back up. Not sure why I bent it down in the first place.
